ddtrace/contrib/botocore/patch.py

```diff
@@ -52,23 +53,87 @@ def inject_trace_data_to_message_attributes(trace_data, entry):
         log.debug("skipping trace injection, max number (10) of MessageAttributes exceeded")
 
 
-def inject_trace_to_sqs_batch_message(args, span):
+def inject_trace_to_sqs_or_sns_batch_message(args, span):
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
-    for entry in params["Entries"]:
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
```

Comment:
I believe the logic is grab the entries, if they are not there then try to grab the PublishBatchRequestEntries and if they are not there then just return an empty list? Why do you do it in that order?

Reply:
Also what even is an "entry" in this context?

Reply:
An entry here is an SNS or SQS record, and depending on how it was published, it could either show up under `Entries` (in case of `Publish`, `PutRecord` & `PutRecords`), or `PublishBatchRequestEntries` (in case of `PublishBatch`). The order doesn't really matter here, but `Entries` is more likely than `PublishBatchRequestEntries` so it makes sense to check there first.

Reply:
Added a comment about this to clarify, I think the one liner is good here with the comment explaining the logic.